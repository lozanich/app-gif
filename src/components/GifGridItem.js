import React from "react";
import PropsTypes from "prop-types";

function GifGridItem({ id, title, url, deleteImageG }) {
  return (
    <div className="card animate__animated animate__bounce">
      <img src={url} alt={title} />
      <p>{title}</p>
      <button onClick={() => deleteImageG(id)} className="button-primary">
        Eliminar
      </button>
    </div>
  );
}

GifGridItem.propTypes = {
  id: PropsTypes.string.isRequired,
  title: PropsTypes.string.isRequired,
  url: PropsTypes.string.isRequired,
};

export default GifGridItem;
