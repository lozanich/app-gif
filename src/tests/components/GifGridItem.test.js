import "@testing-library/jest-dom";
import React from "react";
import GifGridItem from "../../components/GifGridItem";
import { shallow } from "enzyme";

describe("Pruebas al GridItem Component", () => {
  const deleteImage = (id) => {};
  const img = {
    id: "12jhsa7",
    title: "Goku",
    url: "https://",
  };
  const wrapper = shallow(
    <GifGridItem
      id={img.id}
      title={img.title}
      url={img.url}
      deleteImageG={deleteImage}
    />
  );
  // beforeEach(() => {
  //   img = {
  //     id: "12jhsa7",
  //     title: "Goku",
  //     url: "https://",
  //   };

  //   wrapper = shallow(
  //     <GifGridItem
  //       id={img.id}
  //       title={img.title}
  //       url={img.url}
  //       deleteImageG={deleteImage}
  //     />
  //   );
  // });

  test("Debe renderizar el componente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de tener un parrafo con el titulo", () => {
    expect(wrapper.find("p").text().trim()).toBe(img.title);
  });

  test("Debe utilizar los atributos en la imagen", () => {
    const image = wrapper.find("img");
    expect(image.prop("src")).toBe(img.url);
    expect(image.props().alt).toBe(img.title);
  });

  test("Debe tener animated animate__bounce", () => {
    const div = wrapper.find("div").hasClass("animate__bounce");
    expect(div).toBe(true);
  });
});
