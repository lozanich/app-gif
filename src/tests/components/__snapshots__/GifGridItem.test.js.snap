// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al GridItem Component Debe renderizar el componente 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="Goku"
    src="https://"
  />
  <p>
    Goku
  </p>
  <button
    className="button-primary"
    onClick={[Function]}
  >
    Eliminar
  </button>
</div>
`;
